trl/trainer/dpo_trainer.py

```diff
@@ -290,7 +300,217 @@ def _prepare_deepspeed(self, model: PreTrainedModelWrapper):
         model.eval()
         return model
 
-    def concatenated_inputs(self, batch: Dict[str, Union[List, torch.LongTensor]]) -> Dict[str, torch.LongTensor]:
+    def get_train_dataloader(self) -> DataLoader:
+        # tokenize the dataset and compute reference logps for training datasets
+        self.train_dataset = self.train_dataset.map(self.tokenize_batch_element)
+        if self.precompute_ref_logps:
+            self.train_dataset = self.train_dataset.map(
```

Comment:
Do you happen to know what the answer to the above question is (ie do we run inference on CPU or GPU)?

Reply:
It should run wherever the `ref_model` is at this point, which is already prepared via `accelerate` no?

Reply:
yes however if there is explicitly a ref_model then (since its no being passed to the accelerate we have to move it to the accelerate device) which is what i am doing